data_mappings.bal

```diff
@@ -45,14 +46,270 @@ isolated function getBatchEventJson(BatchEvent batchEvent) returns json {
+    json idEH = check idElementEH.id.id; 
+    revokePublisher.id = idEH.toString();
+    json titleElementEH = check xmldata:toJson(xmlContent/<title>);
+    json titleEH = check titleElementEH.title.title; 
+    revokePublisher.title = titleEH.toString();
+    json updatedElementEH = check xmldata:toJson(xmlContent/<updated>);
+    json updatedEH = check updatedElementEH.updated.updated;
+    revokePublisher.updated = updatedEH.toString();
+    xml content = xmlContent/<content>;
+    RevokePublisherDescription revokePublisherDescription = {};
+    revokePublisherDescription = check mapXmlToRevokePublisherDescriptionRecord(content);
+    revokePublisher.revokePublisherDescription = revokePublisherDescription;
+    return revokePublisher;
+}
+
+# Convert xml content to RevokePublisherDescription record
+#
+# + xmlContent - xml content
+# + return - Return RevokePublisherDescription record or error
+isolated function mapXmlToRevokePublisherDescriptionRecord(xml xmlContent) returns RevokePublisherDescription|error {
+    xmlns "http://schemas.microsoft.com/netservices/2010/10/servicebus/connect";
+    xml description = xmlContent/<RevokedPublisherDescription>;
+    RevokePublisherDescription revokePublisherDescription = {};
+    json NameElement = check xmldata:toJson(description/<Name>);
+    json Name = check NameElement.Name.Name;
+    revokePublisherDescription.Name = Name.toString();
+    return revokePublisherDescription;
+}
+
+# Convert xml content to ConsumerGroup stream
+#
+# + xmlPayload - xml content
+# + return - Return ConsumerGroup stream or error
+isolated function mapToConsumerGroupStream(xml xmlPayload) returns stream<ConsumerGroup>|error {
+    xmlns "http://www.w3.org/2005/Atom";
+    ConsumerGroup[] consumerGroups = [];
+    xml entries = xmlPayload/<entry>;
+    foreach xml entry in entries {
+        ConsumerGroup consumerGroup = {};
+        consumerGroup = check mapXmlToConsumerGroupRecord(entry);
+        consumerGroups.push(consumerGroup);
+    }
+    return consumerGroups.toStream();
+}
+
+# Convert xml content to ConsumerGroup record
+#
+# + xmlContent - xml content
+# + return - Return ConsumerGroup record or error
+isolated function mapXmlToConsumerGroupRecord(xml xmlContent) returns ConsumerGroup|error {
+    xmlns "http://www.w3.org/2005/Atom";
+    ConsumerGroup consumerGroup = {};
+    json idElementEH = check xmldata:toJson(xmlContent/<id>);
+    json idEH = check idElementEH.id.id; 
+    consumerGroup.id = idEH.toString();
+    json titleElementEH = check xmldata:toJson(xmlContent/<title>);
+    json titleEH = check titleElementEH.title.title; 
+    consumerGroup.title = titleEH.toString();
+    json publishedElementEH = check xmldata:toJson(xmlContent/<published>);
+    json publishedEH = check publishedElementEH.published.published;
+    consumerGroup.published = publishedEH.toString();
+    json updatedElementEH = check xmldata:toJson(xmlContent/<updated>);
```

Comment:
We can access value of xml element tags by <path to leaf element>/* 
Shall we check on that approach?

Reply:
Resolved in 6c77321 . Thank you.